Sources/OpenAPIRuntime/Base/OpenAPIMIMEType.swift

```diff
@@ -0,0 +1,170 @@
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftOpenAPIGenerator project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftOpenAPIGenerator project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+import Foundation
+
+/// A container for a parsed, valid MIME type.
+@_spi(Generated)
+public struct OpenAPIMIMEType: Equatable {
+
+    /// The kind of the MIME type.
+    public enum Kind: Equatable {
+
+        /// Any, spelled as `*/*`.
+        case any
+
+        /// Any subtype of a concrete type, spelled as `type/*`.
+        case anySubtype(type: String)
+
+        /// A concrete value, spelled as `type/subtype`.
+        case concrete(type: String, subtype: String)
+
+        public static func == (lhs: Kind, rhs: Kind) -> Bool {
+            switch (lhs, rhs) {
+            case (.any, .any):
+                return true
+            case let (.anySubtype(lhsType), .anySubtype(rhsType)):
+                return lhsType.lowercased() == rhsType.lowercased()
+            case let (.concrete(lhsType, lhsSubtype), .concrete(rhsType, rhsSubtype)):
+                return lhsType.lowercased() == rhsType.lowercased()
+                    && lhsSubtype.lowercased() == rhsSubtype.lowercased()
+            default:
+                return false
+            }
+        }
+    }
+
+    /// The kind of the MIME type.
+    public var kind: Kind
+
+    /// Any optional parameters.
+    public var parameters: [String: String]
+
+    /// Creates a new MIME type.
+    /// - Parameters:
+    ///   - kind: The kind of the MIME type.
+    ///   - parameters: Any optional parameters.
+    public init(kind: Kind, parameters: [String: String] = [:]) {
+        self.kind = kind
+        self.parameters = parameters
+    }
+
+    public static func == (lhs: OpenAPIMIMEType, rhs: OpenAPIMIMEType) -> Bool {
+        guard lhs.kind == rhs.kind else {
+            return false
+        }
+        // Parameter names are case-insensitive, parameter values are
+        // case-sensitive.
+        guard lhs.parameters.count == rhs.parameters.count else {
+            return false
+        }
+        func normalizeKeyValue(key: String, value: String) -> (String, String) {
+            (key.lowercased(), value)
+        }
+        let normalizedLeftParams = Dictionary(
+            uniqueKeysWithValues: lhs.parameters.map(normalizeKeyValue)
+        )
+        let normalizedRightParams = Dictionary(
+            uniqueKeysWithValues: rhs.parameters.map(normalizeKeyValue)
+        )
+        return normalizedLeftParams == normalizedRightParams
+    }
+}
+
+extension OpenAPIMIMEType.Kind: LosslessStringConvertible {
+    public init?(_ description: String) {
+        let typeAndSubtype =
+            description
+            .split(separator: "/")
```

Comment:
Nit: could use maxSplits here.

Reply:
Possibly, but I think that'd complicate us rejecting `foo/bar/baz`, which gets filtered out by the next line where we check exactly for 2 components. We equally want to drop `foo` and `foo/bar/baz`, only accept `foo/bar`.

Reply:
OK, fair enough.